common/types.go

```diff
@@ -573,3 +573,24 @@ func ZeroAddress(nodeLocation Location) Address {
 	internal := InternalAddress{nodeLocation.BytePrefix()}
 	return Address{&internal}
 }
+
+// GenerateLocations generates a logical sequence of locations
+func GenerateLocations(maxRegions, zonesPerRegion int) []Location {
```

Comment:
This is good, but I think one additional helper would be better: `GenerateLocationsForExpansion(expansionNumber int)`

This helper would determine the number of regions and zones according to the equation given in QIP<insert QIP number here>, and then call `GenerateLocations(...)` with those parameters.

Thoughts?

Reply:
admittedly unrelated to this PR... maybe keep in mind though as you and shreekara work through the expansion logic

Reply:
Yes that's a good idea will definitely keep in mind and it will have to be implemented along with expansion otherwise it won't work.